### DISCUSSION TITLE
Kiali does not show jaeger traces

### DISCUSSION THREAD

**Discussion by sterziev88** [kiali/kiali#7324] [Q&A]
### Describe the bug

I have installed Kiali operator via helm chart. I am able to see workload, services, applications and graphs for them but I am not able to see jaeger traces.
![image](https://github.com/kiali/kiali/assets/66157650/1ab49d4c-d4a1-4f43-8c91-c703c4cd0dd8)

That is my configuration:

>         server:
          observability:
            tracing:
              collector_type: otel
              collector_url: http://jaeger-operator-jaeger-collector.jaeger-operator.svc.cluster.local:4317
              enabled: false
              otel:
                ca_name: ''
                protocol: grpc
                skip_verify: true
                tls_enabled: false

>           tracing:
            enabled: true
            use_grpc: true
            provider: "jaeger"
            in_cluster_url: "http://jaeger-operator-jaeger-query..jaeger-operator.svc.cluster.local:16685/jaeger"
            url: "http://jaeger.xxxxxxx"

### Expected Behavior
To be able to see jaeger traces

### Environment
*Learn about how to determine versions [here](https://kiali.io/docs/faq/general/#how-do-i-determine-what-version-i-am-running).*

- **Kiali version:*1.81.0*
- **Istio version:*1.21*
- **Kubernetes version:*1.29*

**Comment by jmazzitelli** [comment]
(_converted to Q&A Discussion as this is a question on configuration, not a bug.  Also, this is a question about configuring the server, not the operator_)

**Comment by jmazzitelli** [comment]
Maybe the docs can help:

* https://kiali.io/docs/configuration/p8s-jaeger-grafana/tracing/jaeger/
* the Kiali CR Jaeger config reference docs: https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.tracing
* the tracing FAQ: https://kiali.io/docs/faq/distributed-tracing/

**Comment by sterziev88** [comment]
Hi, I have rad the documents but everything is set as described in them and still I don't see any traces.

Regarding: [https://kiali.io/docs/configuration/p8s-jaeger-grafana/tracing/jaeger/](url)
I have this:

>      
        external_services:
          tracing:
            enabled: true
            use_grpc: true
            provider: "jaeger"
            in_cluster_url: "http://jaeger-operator-jaeger-query..jaeger-operator.svc.cluster.local:16685/jaeger"
            url: "http://jaeger.xxxxxxx"
and I am able to see the link to my jaeger UI but nothing related to traces on kiali UI.

Regarding: [https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.tracing](url)
Everything which is need is set properly according provided documentation.

Regarding: [https://kiali.io/docs/faq/distributed-tracing/](url)
I don't receive any errors or logs with warnings/error.

Could someone help with this?

**Reply by jshaughn** [reply]
I assume you are seeing traces via the Jaeger UI, so the traces exist?  Anything in the Kiali logs?  Any indication in the Kiali masthead that Jaeger is unreachable?  Are you sure the in_cluster_url is correct?

**Comment by sterziev88** [comment]
Hi @jshaughn Yes, I am able to see the traces via Jaeger UI. In the Kiali logs I see only this regarding jaeger -> **2024-05-08T10:05:22Z INF Create jaeger GRPC client jaeger-operator-jaeger-query.jaeger-operator.svc.cluster.local:16685**.
I am using an opentelemetry collector in order to collect traces from the applications ( I am not sure if that can be related ) but after that I send them to jaeger to that endpoint **jaeger-operator-jaeger-collector.jaeger-operator.svc.cluster.loca**l  and then I use query service of jeager so Yes, that should be the correct in_cluster_url -> **jaeger-operator-jaeger-query.jaeger-operator.svc.cluster.local**

**Reply by jshaughn** [reply]
Kiali is only consuming the traces, so using OTel as a collector is fine and should not be relevant.

Do you have a very low number of traces?  Perhaps increase the sampling by adjusting the `Limit per query` setting in the `Display` dropdown?

Is it only the Traces tab on the workload detail page?  If you select a node in the traffic graph, do you see anything under the `Traces` tab in the side panel?

![image](https://github.com/kiali/kiali/assets/2104052/041f40ed-a551-4a85-becc-f291a87c9451)

@josunect , do you have any ideas here?

**Reply by josunect** [reply]
I wonder if the issue is related to some information missing in the trace. 
The setting `tracing.namespace_selector` is true but default, could you try setting that to false and see if it can find any traces?

**Comment by sterziev88** [comment]
Hi @jshaughn and @josunect I set `tracing.namespace_selector` to false and now I am able to see traces.
![image](https://github.com/kiali/kiali/assets/66157650/20f41bb0-795d-47c9-83a0-7ca3d7209766)
I am wondering why this change fix the things.
Thank you very much both of you!

**Reply by josunect** [reply]
Thanks for let us know!
This is because, Kiali queries the trace service by the service name (Ex productpage-bookinfo), and it needs to match exactly. This is usually the service name and the namespace, but sometimes it is just stored without it (Ex productpage).

**Comment by gkotsev** [comment]
Hi, I have similar Kiali issue, but the problem for me is with the Inbound / Outbound Metrics. I'm able to see metrics in Grafana, but nothing in Kiali:
<img width="1282" alt="image" src="https://github.com/kiali/kiali/assets/116803604/3b377589-f807-4128-84ca-c601c2094b7f">

Grafana is on a remote cluster and this is the configuration I have:

```
grafana:
  auth:
    type: "basic"
    username: "?????"
    password: "?????"
  dashboards:
  - name: "Istio Service Dashboard"
    variables:
      namespace: "var-namespace"
      service: "var-service"
  - name: "Istio Workload Dashboard"
    variables:
      namespace: "var-namespace"
      workload: "var-workload"
  - name: "Istio Mesh Dashboard"
  - name: "Istio Control Plane Dashboard"
  - name: "Istio Performance Dashboard"
  - name: "Istio Wasm Extension Dashboard"
  enabled: true
  in_cluster_url: "http://grafana.???.com"
  url: "http://grafana.???.com
```

**Reply by josunect** [reply]
This information is really obtained from Prometheus, the configuration for Grafana is used to redirect to the right page for the Istio Dashboards in Grafana.
I wonder if you are able to see the Traffic graph?

**Reply by gkotsev** [reply]
Hi @josunect, yes, there is information in the Traffic graph:
<img width="909" alt="image" src="https://github.com/kiali/kiali/assets/116803604/942bf1e3-cbfc-4eb5-ad8c-e9697bd681ea">
I'm also able to see info in the Traffic section for the applications:
<img width="948" alt="image" src="https://github.com/kiali/kiali/assets/116803604/03843cec-b30e-4254-9553-eeb8f9dda5c3">

**Reply by josunect** [reply]
Thanks, @gkotsev 

Is this from the same cluster? 
If you follow the links: 
Traffic > View metrics
![image](https://github.com/kiali/kiali/assets/49480155/c86b8a5b-11b0-4167-8eb0-b0b8957d1c08)
Is there any data? 
If you follow the Grafana link (In the right corner): 
![image](https://github.com/kiali/kiali/assets/49480155/9f262253-070e-422d-8c2c-b3af5456a1a8)

Do you see any data in the dashboard? Is the information correct? (The workload, and namespace)
![image](https://github.com/kiali/kiali/assets/49480155/fe20df95-0ffe-458d-bca2-ede0c4f076fc)

**Reply by gkotsev** [reply]
Hi @josunect. 
Yes, this is the same cluster.
It looks like I don't have the action link "View metrics":
<img width="748" alt="image" src="https://github.com/kiali/kiali/assets/116803604/e06b314b-49a2-4026-8d51-29e0c31bcde7">
But when I follow the Grafana link, I can see some graphs:
<img width="1468" alt="image" src="https://github.com/kiali/kiali/assets/116803604/02dcb390-a355-45b2-9876-02fa23e1a39a">

<img width="1472" alt="image" src="https://github.com/kiali/kiali/assets/116803604/6c5ea0d0-8ce7-4d63-9801-bb6a459a9218">

**Reply by josunect** [reply]
Thanks, @gkotsev 
When you follow the Grafana link from Kiali, is this information correct? (Workload, namespace)

![image](https://github.com/kiali/kiali/assets/49480155/2bbdbf97-7c4f-4f41-89cb-e3daaffa07cb)

**Reply by gkotsev** [reply]
Yes, it's the same Workload, Namespace and Reporter.

**Reply by josunect** [reply]
Kiali uses the following [metrics](https://kiali.io/docs/faq/general/#requiredmetrics) to calculate these graphs.
Setting the Kiali log level to "trace", will print the exact queries that Kiali is doing to Prometheus, this is an example: 

`TRC [Prom] fetchRange: sum(rate(istio_tcp_connections_opened_total{reporter="source",source_workload_namespace="bookinfo",source_workload="productpage-v1",source_cluster="Kubernetes"}[36s])) by (destination_canonical_service,destination_workload_namespace)`

This can be very useful (You can use the exactly metric in Prometheus) to troubleshoot, and if some metrics are missing. 

This is a very useful message at the end: 
`TRC [Prom] GetExistingMetricNames: exec time=[12.451443ms], results count=[1160], looking for count=[10], found count=[10]`
